drv/lpc55-update-server/src/main.rs

```diff
@@ -288,6 +287,26 @@ fn do_block_write(
     }
 }
 
+fn do_block_erase(
+    img: UpdateTarget,
+    block_num: usize,
+) -> Result<(), UpdateError> {
+    let result = unsafe {
+        // The write_to_flash API takes raw pointers due to TrustZone
+        // ABI requirements which makes this function unsafe.
+        tz_table!().erase_flash(img, block_num as u32)
+    };
+
+    match result {
```

Comment:
Now that this is used in two places, we should probably `impl From<HypoStatus> for Result<(), UpdateError>`